webapp/portlet/src/main/webapp/vue-apps/favorites-list-top-bar/components/ActivityFavoriteItem.vue

```diff
@@ -32,13 +38,38 @@ export default {
       type: String,
       default: () => null,
     },
+    activityExtensions: {
+      type: Object,
+      default: () => null,
+    }
   },
   data: () => ({
     activity: null,
     activityTitle: '',
     activityUrl: '#',
-    isFavorite: true
+    isFavorite: true,
+    defaultIcon: 'fas fa-feather-alt',
+    defaultClass: 'primary--text',
+    activityTypeExtension: {},
   }),
+  watch: {
+    activity() {
+      if (this.activity?.type) {
+        this.activityExtensions.forEach(extension => {
+          if (extension.type === this.activity.type) {
+            this.activityTypeExtension.icon = extension?.icon ? extension.icon : '';
+            this.activityTypeExtension.img = extension?.img ? extension.img : '';  
+            this.activityTypeExtension.class = extension?.class ? extension.class : '';
+            this.activityTypeExtension.title = extension?.title ? extension.title(this.activity) : this.$t('UITopBarFavoritesPortlet.label.activity');
+          }
+        });
+      } else {
+        this.activityTypeExtension.icon = this.defaultIcon;
+        this.activityTypeExtension.class = this.defaultClass;
+        this.activityTypeExtension.title = this.activity?.title ? this.favoriteTitle(this.activity.title) : this.$t('UITopBarFavoritesPortlet.label.activity');
+      }
+    }
+  },
   computed: {
```

Comment:
Delete watch and add it as computed : 

```suggestion
  }),
  computed: {
    activityTypeExtension() {
      if (this.activity?.type) {
        return this.activityExtensions.find(extension => extension.type === this.activity.type);
      }
      return null;
    },
    activityTitle() {
      return this.activityTypeExtension?.title && this.activityTypeExtension?.title(this.activity) || this.favoriteTitle(this.activity.title) || this.$t('UITopBarFavoritesPortlet.label.activity');
    }
  },
```

Reply:
done